subprojects/core/src/main/java/org/gradle/api/provider/ProviderFactory.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.provider;
```

Comment:
The interface `ProviderFactory` is public because you can inject into classes via `javax.inject.Inject` e.g. see test case `ProviderIntegrationTest.can inject and use provider factory via annotation`. Let me know if you had something else in mind.